config_test.go

```diff
@@ -207,9 +207,98 @@ func TestSettingUpdated(t *testing.T) {
 
 	require.NoError(t, conf.Set(DeviceKey, randomDevice()))
 	require.Equal(t, conf.Get(DeviceKey+".updated"), now())
+}
 
-	require.NoError(t, conf.Set(DeviceKey+".name", randString(10)))
-	require.Equal(t, conf.Get(DeviceKey+".updated"), now())
+func TestMapToStruct(t *testing.T) {
+	defer newFs(t, "")()
+
+	conf, err := New(DefaultConfigDir)
+	require.NoError(t, err)
+
+	newNow()
+	locationMap := map[string]interface{}{
+		"latitude":  "123.321",
+		"timestamp": now().Format(TimeFormat),
+	}
+	locationExpected := Location{
+		Latitude:  123.321,
+		Timestamp: now(),
+	}
+	var location Location
+	require.NoError(t, conf.SetFromMap(LocationKey, locationMap))
+	require.NoError(t, conf.Unmarshal(LocationKey, &location))
+	equalLocation(t, locationExpected, location)
+
+	audioMap := map[string]interface{}{
+		"directory":      "/audio/directory",
+		"card":           "4",
+		"volume-control": "audio volume control",
+	}
+	audioExpected := Audio{
+		Dir:           "/audio/directory",
+		Card:          4,
+		VolumeControl: "audio volume control",
+	}
+	checkWritingMap(t, AudioKey, &Audio{}, &audioExpected, audioMap, conf)
+
+	batteryMap := map[string]interface{}{"enable-voltage-readings": "true"}
+	batteryExpected := Battery{EnableVoltageReadings: true}
+	checkWritingMap(t, BatteryKey, &Battery{}, &batteryExpected, batteryMap, conf)
+
+	deviceMap := map[string]interface{}{"Group": "a-group"}
+	deviceExpected := Device{Group: "a-group"}
+	checkWritingMap(t, DeviceKey, &Device{}, &deviceExpected, deviceMap, conf)
+
+	modemdMap := map[string]interface{}{
+		"test-interval": "10m4s",
+		"modems": []map[string]interface{}{
+			map[string]interface{}{
+				"name": "modem name",
+			},
+		},
+	}
+	modemdExpected := Modemd{
+		TestInterval: 10*time.Minute + 4*time.Second,
+		Modems:       []Modem{Modem{Name: "modem name"}},
+	}
+	checkWritingMap(t, ModemdKey, &Modemd{}, &modemdExpected, modemdMap, conf)
+}
```

Comment:
These tests are really great but please split them up so there's one for each config section.